homeassistant/components/xiaomi_miio/vacuum.py

```diff
@@ -487,10 +487,10 @@ def update(self):
             self._timers = self._vacuum.timer()
 
             self._available = True
-        except OSError as exc:
-            _LOGGER.error("Got OSError while fetching the state: %s", exc)
-        except DeviceException as exc:
-            _LOGGER.warning("Got exception while fetching the state: %s", exc)
+        except (OSError, DeviceException) as exc:
+            if self._available:
+                self._available = False
+                _LOGGER.warning("Got exception while fetching the state: %s", exc)
```

Comment:
Secondly, if we want to test that this warning occurs, we need to first make sure that the entity is available before the error happens. Right now the entity is not available on creation so this log message won't happen.